robot/impl/local_robot_test.go

```diff
@@ -221,7 +404,42 @@ func TestConfigRemote(t *testing.T) {
 
 	fsConfig, err := r2.FrameSystemConfig(context.Background())
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, fsConfig.Parts, test.ShouldHaveLength, 12)
+	test.That(t, fsConfig.Parts, test.ShouldHaveLength, 7)
+
+	// Assert that a client connected to r2 gets the appropriate resource names from
+	// ResourceNames. No `Remote` fields are set, and all prefixes are correctly applied.
+	options, _, r2Addr := robottestutils.CreateBaseOptionsAndListener(t)
+	err = r2.StartWeb(ctx, options)
+	test.That(t, err, test.ShouldBeNil)
+
+	r2Client, err := client.New(ctx, r2Addr, logger.Sublogger("client"))
+	test.That(t, err, test.ShouldBeNil)
+	defer func() {
+		test.That(t, r2Client.Close(ctx), test.ShouldBeNil)
+	}()
+
+	expectedResourceNamesFromClient := []resource.Name{
```

Comment:
This is the expected slice of `resource.Name` returned by calling `ResourceNames()` _through an SDK_. So, it's NOT expected that `"myremote:..."` prepensions will be present.